crates/parcel/benches/parcel_benchmarks.rs

```diff
@@ -0,0 +1,162 @@
+//! This benchmark measures write performance of individual request results into an LMDB cache
+//! backend using different serialization and write strategies.
+use std::sync::atomic::{AtomicU64, Ordering};
+
+use criterion::{black_box, criterion_group, criterion_main, BatchSize, Criterion};
+
+use parcel::cache::{LMDBCache, LMDBCacheOptions};
+use parcel::requests::asset_request::AssetRequestOutput;
+use parcel::requests::RequestResult;
+use parcel_core::cache::Cache;
+use parcel_core::types::Asset;
+
+struct BenchmarkItem {
+  request_result: RequestResult,
+  cache_key: String,
+}
+
+static CURRENT_KEY: AtomicU64 = AtomicU64::new(0);
+
+fn setup() -> BenchmarkItem {
+  let asset = Asset::default();
+  let request_result = RequestResult::Asset(AssetRequestOutput {
+    asset,
+    dependencies: vec![],
+  });
+  let cache_key = CURRENT_KEY.fetch_add(1, Ordering::Relaxed).to_string();
+  BenchmarkItem {
+    request_result,
+    cache_key,
+  }
+}
+
+/// Benchmark writes and serialization to LMDB
+pub fn cache_benchmark(c: &mut Criterion) {
+  let cache = LMDBCache::new(Default::default()).unwrap();
+
+  c.bench_function("serialize request using bincode to_bytes", |b| {
+    let mut output = vec![];
+    output.resize(10000, 0); // pre-alloc
+    b.iter_batched(
+      setup,
+      |BenchmarkItem { request_result, .. }| {
+        bincode::encode_into_slice(&request_result, &mut output, bincode::config::standard())
+          .unwrap();
+        black_box(&output);
+      },
+      BatchSize::SmallInput,
+    );
+  });
+
+  benchmark_suite(c, "async writes", cache);
+
+  let cache = LMDBCache::new(LMDBCacheOptions {
+    async_writes: false,
+    ..Default::default()
+  })
+  .unwrap();
+  benchmark_suite(c, "sync writes", cache);
+}
+
+fn benchmark_suite(c: &mut Criterion, name: &str, cache: LMDBCache) {
+  let _ = std::fs::remove_dir_all(".parcel-cache/rust-cache").unwrap();
```

Comment:
This should probably use `.join` along with the other usages, otherwise it won't be cross-platform